_includes/markdown/JavaScript.md

```diff
@@ -103,7 +103,23 @@ Certain codebases may already have such libraries in place. Many legacy projects
 ### Concatenating Requests
 When constructing a page that contains a lot of client-side data requests you will want to consider concatenating your requests into a single Ajax call. This will help you avoid piling up requests or sending them through callbacks and nested promises when parts of the data depend on other parts.
 
-[GraphQL](https://graphql.org/) is an open source query language that can help with this situation by allowing you to combine multiple API requests into a single call. In a WordPress environment, the [WPGraphQL](https://github.com/wp-graphql/wp-graphql) plugin will let you directly access the WordPress JSON API.
+#### GraphQL
+[GraphQL](https://graphql.org/) is an open source data query and manipulation language. It provides an alternative to REST because it allows for a consistent way to make declarative queries. We first define the data structure(s) we need, then request the data, and return only the data that was requested. This creates an environment of smaller, more targeted calls to an API. It also allows us to concatenate multiple calls into single data requests, reducing the overhead and time to load.
+
+An essential part of a GraphQL API is an API schema. GraphQL requires a human-readable schema which describes the types which are available, and how they relate to one another. While writing a schema is reasonably straightforward, utilizing the standardized nature of WordPress’s database could save time. It may well be possible to reuse a schema from other projects, such as generated by a plugin like, [WPGraphQL](https://github.com/wp-graphql/wp-graphql).
+
+#### WPGraphQL
+If you are choosing to use GraphQL on a Wordpress project, it is recommended to use the WPGraphQL plugin. This plugin will make all the data in the WordPress JSON API available through a GraphQL endpoint - in many cases you won’t need to write the schema yourself. This will give you all the benefits of concatenating your data requests as well as easy access to your data as it is output by WordPress. You can retrieve your data by passing a query directly into a simple fetch request:
+
+```javascript
+fetch( '/graphql', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ [insert your query here] }),
```

Comment:
So your query will need to look like: 

```
body: JSON.stringify({ query: "[your query string goes here]"})
```

Reply:
with WPGraphQL (or any other GraphQL server that supports Batch requests, you can also send an array of queries like so: 

```
body: JSON.stringify([{ query: "[your query string goes here]"}, {query: "[your next query goes here]"}])
```

Apollo Client can automatically generate and send batch queries for you via Apollo Link Batch HTTP: https://github.com/apollographql/apollo-link/tree/master/packages/apollo-link-batch-http. 

Batch queries can further increase the performance of client-side applications as fewer HTTP requests are needed to get the data necessary for your application. 

This is common when first loading a page or changing routes that are composed of many components that each have their own GraphQL dependencies. 

For example, let's say you had the following: 

```
<Layout>
  <Menu />
  <RecentPosts />
</Layout>
```

The `<Menu>` component could make use of a WPGraphQL `menuItems` query, and the `<RecentPosts />` component could make use of a `posts` query. 

When the page first loads, Apollo Link HTTP Batch would send both the `menuItems` query _and_ the `posts` query in a single HTTP request. 

So you get the benefit of declaratively writing queries for the data you need, coupling the data dependencies with the components that need the data, and having the data fetched in the most efficient way possible. 💥